pkg/scrapers/twitter/auth.go

```diff
@@ -2,35 +2,98 @@ package twitter
 
 import (
 	"fmt"
+	"sync"
+	"time"
 
+	"github.com/masa-finance/masa-oracle/pkg/config"
 	twitterscraper "github.com/masa-finance/masa-twitter-scraper"
+	"github.com/sirupsen/logrus"
 )
 
-// Login attempts to log in to the Twitter scraper service.
-// It supports three modes of operation:
-// 1. Basic login using just a username and password.
-// 2. Login requiring an email confirmation, using a username, password, and email address.
-// 3. Login with two-factor authentication, using a username, password, and 2FA code.
-// Parameters:
-//   - scraper: A pointer to an instance of the twitterscraper.Scraper.
-//   - credentials: A variadic list of strings representing login credentials.
-//     The function expects either two strings (username, password) for basic login,
-//     or three strings (username, password, email/2FA code) for email confirmation or 2FA.
-//
-// Returns an error if login fails or if an invalid number of credentials is provided.
+type TwitterAccount struct {
+	Username         string
+	Password         string
+	TwoFACode        string
+	RateLimitedUntil time.Time
+}
+
+type TwitterAccountManager struct {
+	accounts []*TwitterAccount
+	index    int
+	mutex    sync.Mutex
+}
+
+func NewTwitterAccountManager(accounts []*TwitterAccount) *TwitterAccountManager {
+	return &TwitterAccountManager{
+		accounts: accounts,
+		index:    0,
+	}
+}
+
+func (manager *TwitterAccountManager) GetNextAccount() *TwitterAccount {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	for i := 0; i < len(manager.accounts); i++ {
+		account := manager.accounts[manager.index]
+		manager.index = (manager.index + 1) % len(manager.accounts)
+		if time.Now().After(account.RateLimitedUntil) {
+			return account
+		}
+	}
+	return nil
+}
+
+func (manager *TwitterAccountManager) MarkAccountRateLimited(account *TwitterAccount) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	account.RateLimitedUntil = time.Now().Add(time.Hour)
+}
+
+func Auth(account *TwitterAccount) *twitterscraper.Scraper {
+	scraper := twitterscraper.New()
+	baseDir := config.GetInstance().MasaDir
+
+	if err := LoadCookies(scraper, account, baseDir); err == nil {
+		logrus.Debugf("Cookies loaded for user %s.", account.Username)
+		if IsLoggedIn(scraper) {
+			logrus.Debugf("Already logged in as %s.", account.Username)
+			return scraper
+		}
+	}
+
+	time.Sleep(100 * time.Millisecond)
```

Comment:
constant sleep time over here - shall we have this configurable?

Reply:
@mudler added this to a package config for now: https://github.com/masa-finance/masa-oracle/pull/576/commits/7ad1bfea4bd0482afbe4e4200a168fac9a40c2a2

Reply:
Updated here: https://github.com/masa-finance/masa-oracle/pull/576/commits/1797bfecdf8646966133efd13390d0d95c7c4ba6

I am leaving in the package config.go for now until we structure the config better